src/v2/Apps/Settings/SettingsApp.tsx

```diff
@@ -20,7 +21,23 @@ const TABS = [
   { name: "Shipping", url: "/settings/shipping" },
 ]
 
+// Tabs with My Collection (should be removed when the feature flag is enabled)
+const TABS_WITH_MYC = [
+  { name: "Edit Settings", url: "/settings/edit-settings" },
+  { name: "Collector Profile", url: "/settings/edit-profile" },
+  { name: "My Collection", url: "/settings/my-collection" },
+  { name: "Saves & Follows", url: "/settings/saves" },
+  { name: "Your Alerts", url: "/settings/alerts" },
+  { name: "Order History", url: "/settings/purchases" },
+  { name: "Bids", url: "/settings/auctions" },
+  { name: "Payments", url: "/settings/payments" },
+  { name: "Shipping", url: "/settings/shipping" },
+]
+
 const SettingsApp: React.FC<SettingsAppProps> = ({ me, children }) => {
+  const isMyCollectionEnabled = useFeatureFlag("my-collection-web")
+
+  const tabs = isMyCollectionEnabled ? TABS_WITH_MYC : TABS
```

Comment:
I think instead of using two different arrays you can push a new element to the specific position of the array
Should be something like this
```TABS.splice(2, 0, { name: "My Collection", url: "/settings/my-collection" })```

Reply:
But I don't mind your approach as well :)

Reply:
It was my first thought, but when I tried it with a useEffect there was a weird behavior with it, sometimes it doesn't render, and sometimes it renders after a delay (check video)
but maybe it's my device, maybe it could be solved with SSR

https://user-images.githubusercontent.com/20655703/180221650-882e4eac-c893-4a9a-abaa-1effc082b3dc.mp4

Reply:
You could give [compact](https://lodash.com/docs/4.17.15#compact) a try:

```
const TABS = compact([
  ...
  isMyCollectionEnabled && { name: "My Collection", url: "/settings/my-collection" },
  ...
])
```